nvi-rest/src/test/java/no/sikt/nva/nvi/rest/upsert/UpdateNviCandidateStatusHandlerDbTest.java

```diff
@@ -63,22 +66,51 @@ void shouldUpdateApprovalStatus(NviApprovalStatus status) throws IOException {
                                                                   .status(DbStatus.PENDING)
                                                                   .build()));
 
-        var req = new NviStatusRequest(candidate.identifier(), institutionId, status);
-        var request = createRequest(req, institutionId);
+        var req = new NviStatusRequest(institutionId, status);
+        var request = createRequest(req, institutionId, candidate.identifier());
         handler.handleRequest(request, output, context);
         var gatewayResponse = GatewayResponse.fromOutputStream(output, CandidateResponse.class);
         var bodyAsInstance = gatewayResponse.getBodyObject(CandidateResponse.class);
         assertThat(bodyAsInstance.approvalStatuses().get(0).status().getValue(), is(equalTo(status.getValue())));
     }
 
+    @Test
+    void shouldUpdateRejectedApprovalToPending() throws IOException {
+        var institutionId = randomUri();
+        var dbCandidate = createCandidate(institutionId);
+        var candidate = nviCandidateRepository.create(dbCandidate,
+                                                      List.of(DbApprovalStatus.builder()
+                                                                  .institutionId(institutionId)
+                                                                  .status(DbStatus.PENDING)
+                                                                  .build()));
+        nviService.updateApprovalStatus(candidate.identifier(), rejectedApproval(candidate));
+        var request = createRequest(createRequestBody(institutionId, NviApprovalStatus.PENDING),
+                                    institutionId,
+                                    candidate.identifier());
+        handler.handleRequest(request, output, context);
+        var gatewayResponse = GatewayResponse.fromOutputStream(output, CandidateResponse.class);
+        var bodyAsInstance = gatewayResponse.getBodyObject(CandidateResponse.class);
+        assertThat(bodyAsInstance.approvalStatuses().get(0).status().getValue(),
+                   is(equalTo(NviApprovalStatus.PENDING.getValue())));
+    }
+
+    private static NviStatusRequest createRequestBody(URI institutionId, NviApprovalStatus approvalStatus) {
+        return new NviStatusRequest(institutionId, approvalStatus);
+    }
+
+    private static DbApprovalStatus rejectedApproval(Candidate candidate) {
+        return candidate.approvalStatuses().get(0).copy()
```

Comment:
In another PR, we need to be able to easily access a given approval in a sensible way…i.e. what is a unique and relevant way of accessing an approval? Probably, `Optional<ApprovalStatus> approvalStatuses.getApprovalForCustomer(URI customer)`